topics/integrate-in-existing-app.md

```diff
@@ -1,242 +1,415 @@
 
-You can also choose to share the UI behavior using the [Model-View-Presenter (_MVP_)](#mvp-for-legacy-ui-frameworks) 
-or the [Model-View-Intent (_MVI_)](#mvi-for-declarative-ui-frameworks) pattern. These patterns:
+   ![KMM shared module](kmm-module-wizard-1.png)
 
-*   Make a clear distinction between the UI and presentation layers.
-*   Are completely decoupled from the UI platform.
-*   Are easily testable without a UI environment.
+3. Click **Finish**.
 
-#### MVP for legacy UI frameworks {initial-collapse-state="collapsed"}
+The wizard will create the KMM shared module, update the configuration files, and create files with classes that demonstrate the benefits of Kotlin Multiplatform.
+You can learn more about the [KMM project structure](discover-kmm-project.md).
 
-_Model-View-Presenter_ (_MVP_) forces you to create an API for both the Presenter that receives inputs and the View that displays outputs, allowing 
-you to test each independently.
+### Add a dependency on the shared module to your Android application
 
-Here is an example of a simple MVP presenter:
+To use cross-platform code in your Android application, connect the shared module to it, move the business logic code there, and make this code cross-platform.
 
-```kotlin
-class LoginPresenter {
-    interface View {
-        fun displayError(code: Int)
-        fun displayLoading(loading: Boolean)
-        fun goToNextScreen()
-    }
+1. Ensure that `compileSdkVersion` and `minSdkVersion` in `build.gradle.kts` of the `shared` module are the same as those in the `build.gradle` of your Android application in the `app` module.  
+   If they are different, update them in the `build.gradle.kts` of the shared module. Otherwise, you'll encounter a compile error.
 
-    private var view: View? = null
-    private var lastCommand: View.() -> Unit =
-            { displayLoading(false) }
-    private fun commandView(command: View.() -> Unit) {
-        lastCommand = command
-        view?.command()
-    }
+2. Add a dependency on the shared module to the `build.gradle` of your Android application.
 
-    fun attach(view: View) { this.view = view.apply(lastCommand) }
-    fun detach() { this.view = null }
-
-    fun login(username: String, password: String) {
-        MainScope().launch {
-            try {
-                commandView { displayLoading(true) }
-                getNetwork().login(username, password) // suspending
-                commandView { goToNextScreen() }
-            } catch (ex: LoginException) {
-                commandView {
-                    displayLoading(false)
-                    displayError(ex.code)
-                }
-            }
-        }
+    ```kotlin
+    dependencies {
+        implementation project(':shared')
     }
-}
-```
+    ```
 
-Note the `commandView` and `lastCommand` mechanism, which allows a view to detach and re-attach, for example for configuration changes on Android.
+3. Synchronize the project to apply the changes.
```

Comment:
minor: Here and below, many actions require pressing buttons. Some cases are described explicitly, and some are not.
I guess this is a matter of saving space, but it looks a little inconsistent.

Reply:
I've added here. As to other places, I tried to balance a description and describe explicitly steps where the user may go a wrong way. But you're right it's not consistent. I don't see a big problem if a user follows steps and gets a result. It's a tutorial, not a procedure in docs. I would keep it flexible and listen to feedback.